geode-core/src/main/java/org/apache/geode/distributed/internal/ClusterConfigurationService.java

```diff
@@ -352,6 +358,20 @@ public boolean removeJars(final String[] jarNames, String[] groups) {
         if (configuration == null) {
           break;
         }
+
+        for (String jarRemoved : jarNames) {
```

Comment:
we don't need to do this if we remove the isOriginRemote() check in the listener. Essentially we want all the updates to touch just the region entry, while letting the listener deal with the file system update.